rubicon/objc/api.py

```diff
@@ -1554,42 +1552,47 @@ def __subclasscheck__(self, subclass):
 NSObjectProtocol = ObjCProtocol('NSObject')
 
 
-# Instances of DeallocationObserver are associated with every
-# ObjCInstance that needs to be kept alive, which is the case
-# when the ObjCInstance has had additional Python attributes added to it.
-# Their sole purpose is to watch for when the Objective-C object
-# is deallocated, and then remove the object from the
-# ObjCInstance._keep_alive_objects dictionary.
+# When a Python object is assigned to a new ObjCInstance attribute, the Python
+# object should be kept alive for the lifetime of the ObjCInstance. This is
+# done by wrapping the Python object as a WrappedPyObject that manages increments
+# the reference count during assignment and decrements it the the WrappedPyObject
+# and the owning ObjCInstance are deallocated.
 #
 # The methods of the class defined below are decorated with
-# rawmethod() instead of method() because DeallocationObservers
+# rawmethod() instead of method() because WrappedPyObject
 # are created inside of ObjCInstance's __new__ method and we have
 # to be careful to not create another ObjCInstance here (which
 # happens when the usual method decorator turns the self argument
 # into an ObjCInstance), or else get trapped in an infinite recursion.
 
-# Try to reuse an existing DeallocationObserver class.
+# Try to reuse an existing WrappedPyObject class.
 # This allows reloading the module without having to restart
-# the interpreter, although any changes to DeallocationObserver
+# the interpreter, although any changes to WrappedPyObject
 # itself are only applied after a restart of course.
 try:
-    DeallocationObserver = ObjCClass("DeallocationObserver")
+    WrappedPyObject = ObjCClass("WrappedPyObject")
 except NameError:
-    class DeallocationObserver(NSObject):
+    class WrappedPyObject(NSObject):
 
-        observed_object = objc_ivar(objc_id)
+        wrapped_pointer = objc_ivar(c_long)
```

Comment:
Would `c_void_p` work here instead, to make it clear that it's a pointer?

Reply:
Yes.